profiler/include/profile_convnd_bwd_data_impl.hpp

```diff
@@ -0,0 +1,439 @@
+                                                      conv_filter_dilations,
+                                                      input_left_pads,
+                                                      input_right_pads,
+                                                      InElementOp{},
+                                                      WeiElementOp{},
+                                                      OutElementOp{});
+            ref_invoker.Run(ref_argument);
+        };
+        switch(NDimSpatial)
+        {
+        case 3: {
+            auto ref_conv = ck::tensor_operation::host::ReferenceConvBwdData<InDataType,
+                                                                             WeiDataType,
+                                                                             OutDataType,
+                                                                             InElementOp,
+                                                                             WeiElementOp,
+                                                                             OutElementOp,
+                                                                             3>();
+            RunReference(ref_conv);
+            break;
+        }
+        case 2: {
+            auto ref_conv = ck::tensor_operation::host::ReferenceConvBwdData<InDataType,
+                                                                             WeiDataType,
+                                                                             OutDataType,
+                                                                             InElementOp,
+                                                                             WeiElementOp,
+                                                                             OutElementOp,
+                                                                             2>();
+            RunReference(ref_conv);
+            break;
+        }
+        case 1: {
+            auto ref_conv = ck::tensor_operation::host::ReferenceConvBwdData<InDataType,
+                                                                             WeiDataType,
+                                                                             OutDataType,
+                                                                             InElementOp,
+                                                                             WeiElementOp,
+                                                                             OutElementOp,
+                                                                             1>();
+            RunReference(ref_conv);
+            break;
+        }
+        default: {
+            throw std::runtime_error("Unsupported number of spatial dimensions provided!");
+        }
+        }
+    }
+
+    // add device Conv instances
+    std::vector<DeviceConvBwdDataNoOpPtr> conv_ptrs;
+    GetDeviceConvBwdDataOpPtr(InDataType{}, conv_ptrs, NDimSpatial);
```

Comment:
`WeiDataType` and `OutDataType` should also be types in `GetDeviceConvBwdDataOpPtr`.